src/agent/testresultreader.ts

```diff
@@ -0,0 +1,404 @@
+            if(rootNode.attributes().timestamp) {
+                var timestampFromXml = new Date(rootNode.attributes().timestamp);
+                if(timestampFromXml < new Date()) {
+                    timeStamp = timestampFromXml;
+                }                    
+            }
+
+            if(rootNode.attributes().time) {
+                totalRunDuration = rootNode.attributes().time; //in seconds
+            }
+
+            //find test case nodes in JUnit result xml
+            var testResults = [];
+
+            for(var i = 0; i < rootNode.testcase.count(); i ++) {
+                var testCaseNode = rootNode.testcase.at(i);
+
+                //testcase name and type
+                var testName = "";
+                if(testCaseNode.attributes().name) {
+                    testName = testCaseNode.attributes().name;                    
+                } 
+
+                var testStorage = "";
+                if(testCaseNode.attributes().classname) {
+                    testStorage = testCaseNode.attributes().classname;
+                }
+
+                //testcase duration
+                var testCaseDuration = 0; //in seconds
+                if(testCaseNode.attributes().time) {
+                    testCaseDuration = testCaseNode.attributes().time;
+                    totalTestCaseDuration = totalTestCaseDuration + testCaseDuration;
+                }
+                
+                //testcase outcome
+                var outcome = "Passed";
+                var errorMessage = "";
+                if(testCaseNode.failure) {
+                    outcome = "Failed";
+                    errorMessage = testCaseNode.failure.text();
+                }
+                else if(testCaseNode.error) {
+                    outcome = "Failed";
+                    errorMessage = testCaseNode.error.text();
+                }
+
+                var testResult : ifm.TestRunResult = <ifm.TestRunResult> {
+                    state: "Completed",
+                    computerName: hostName,
+                    resolutionState: null,
+                    testCasePriority: 1,
+                    failureType: null,
+                    automatedTestName: testName,
+                    automatedTestStorage: testStorage,
+                    automatedTestType: "JUnit",
+                    automatedTestTypeId: null,
+                    automatedTestId: null,
+                    area: null,
+                    owner: buildRequestedFor, 
+                    runBy: buildRequestedFor,
+                    testCaseTitle: testName,
+                    revision: 0,
+                    dataRowCount: 0,
+                    testCaseRevision: 0,
+                    outcome: outcome,
+                    errorMessage: errorMessage,
+                    durationInMs: testCaseDuration * 1000, //convert to milliseconds
+                };
+                
+                testResults.push(testResult);
```

Comment:
No need to make any changes, this is fine.  If we are just iterating through a loop and convert the value to some other type, maybe a "map" function could be used.  Since we are writing in a high level functional language, seems like we can go and use some nice features it provides.  :)